src/Razor/test/Microsoft.VisualStudio.LanguageServer.ContainedLanguage.Test/VirtualDocumentFactoryBaseTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+using Moq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.LanguageServer.ContainedLanguage.Test
+{
+    public class VirtualDocumentFactoryBaseTest
+    {
+        public VirtualDocumentFactoryBaseTest()
+        {
+            ContentTypeRegistry = Mock.Of<IContentTypeRegistryService>();
+            var textBufferFactory = new Mock<ITextBufferFactoryService>();
+            textBufferFactory
+                .Setup(factory => factory.CreateTextBuffer())
+                .Returns(Mock.Of<ITextBuffer>(buffer => buffer.CurrentSnapshot == Mock.Of<ITextSnapshot>() && buffer.Properties == new PropertyCollection() && buffer.ContentType == TestVirtualDocumentFactory.LanguageLSPContentTypeInstance));
+            TextBufferFactory = textBufferFactory.Object;
+
+            var hostContentType = Mock.Of<IContentType>(contentType => contentType.IsOfType(TestVirtualDocumentFactory.HostDocumentContentTypeNameConst) == true);
+            HostLSPBuffer = Mock.Of<ITextBuffer>(textBuffer => textBuffer.ContentType == hostContentType);
+
+            var nonHostLSPContentType = Mock.Of<IContentType>(contentType => contentType.IsOfType(It.IsAny<string>()) == false);
+            NonHostLSPBuffer = Mock.Of<ITextBuffer>(textBuffer => textBuffer.ContentType == nonHostLSPContentType);
+
+            TextDocumentFactoryService = Mock.Of<ITextDocumentFactoryService>();
+        }
+
+        private ITextBuffer NonHostLSPBuffer { get; }
+
+        private ITextBuffer HostLSPBuffer { get; }
+
+        private IContentTypeRegistryService ContentTypeRegistry { get; }
+
+        private ITextBufferFactoryService TextBufferFactory { get; }
+
+        private ITextDocumentFactoryService TextDocumentFactoryService { get; }
+
+        [Fact]
+        public void TryCreateFor_NonHostLSPBuffer_ReturnsFalse()
```

Comment:
```suggestion
        public void TryCreateFor_IncompatibleHostDocumentBuffer_ReturnsFalse()
```

Reply:
Sure

---
In reply to: [452550650](https://github.com/dotnet/aspnetcore-tooling/pull/2201#discussion_r452550650) [](ancestors = 452550650)